neural_compressor/common/tuning_param.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+from enum import Enum, auto
+from typing import Any, List
+
+from pydantic import BaseModel
+
+from neural_compressor.common import logger
+
+
+class ParamLevel(Enum):
+    OP_LEVEL = auto()
+    OP_TYPE_LEVEL = auto()
+    MODEL_LEVEL = auto()
+
+
+class TuningParam:
+    """Define the tunable parameter for the algorithm.
+
+    Example:
+        Class FakeAlgoConfig(BaseConfig):
+            '''Fake algo config.'''.
+
+            params_list = [
+                ...
+                # For simple tunable types, like a list of int, giving
+                # the param name is enough. `BaseConfig` class will
+                # create the `TuningParam` implicitly.
+                "simple_attr"
+
+                # For complex tunable types, like a list of lists,
+                # developers need to create the `TuningParam` explicitly.
+                TuningParam("complex_attr", tunable_type=List[List])
```

Comment:
Hi @mengniwang95, to handle complicated tunable types, such as a list of lists, I would like to introduce this mechanism. Please have a look :).